src/Listener/UserListener.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Listener;
+
+use App\Entity\Client;
+use App\Entity\User;
+use App\Service\CacheService;
+use App\Service\UserService;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Event\PostPersistEventArgs;
+use Doctrine\ORM\Event\PostRemoveEventArgs;
+use Doctrine\ORM\Event\PostUpdateEventArgs;
+use Doctrine\ORM\Event\PreRemoveEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Doctrine\ORM\Mapping as On;
+
+class UserListener
+{
+    public function __construct(
+        private CacheService  $cacheService,
+        private UserService   $userService)
+    {
+    }
+
+    #[On\PreUpdate]
+    public function clearUpdatedCache(User $user, PreUpdateEventArgs $eventArgs)
+    {
+        if ($eventArgs->hasChangedField('Client')) {
+            $oldClients = $eventArgs->getOldValue('Client');
+            $newClients = $eventArgs->getNewValue('Client');
+            $client = $this->findChangedClient($oldClients, $newClients);
+            $this->cacheService->destructCacheByTags(['userList' . $client->getId()]);
+            $this->cacheService->destructCacheByName(($this->userService->cacheNameUserDetail($user->getId(), $client->getId())));
+        }
+        //other changes to users that need cache Clearing can go there
+
+    }
+
+    private function findChangedClient(Collection $oldClients, Collection $newClients): Client
+    {
+        foreach ($oldClients as $client) {
+            if (!$newClients->contains($client)) {
+               return $client;
+            }
+        }
```

Comment:
```suggestion
$modifiedClients = $oldClient->filter(function($client) use $newClients {
        foreach ($oldClients as $client) {
            if (!$newClients->contains($client)) {
               return $client;
            }
        }
return !$newClient->contains($client);
});
```

Idée pour utiliser les options doctrine collection